frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.jsx

```diff
@@ -119,7 +119,7 @@ export class TagEditorParam extends Component {
     if (this.props.tag[attr] !== val) {
       this.props.setTemplateTag({
         ...this.props.tag,
-        [attr]: val,
+        [attr]: val.length > 0 ? val : null,
```

Comment:
[From the failed test](https://www.deploysentinel.com/ci/runs/649ca696d10c320f3efeea9e) `val` could be null that's why this broke the functionality

```suggestion
        [attr]: val?.length > 0 ? val : null,
```

The test tried to clear the default value via the **X** button, but it failed.
![image](https://github.com/metabase/metabase/assets/1937582/a8897eca-6e3d-4f96-b4a3-0c12bba700ca)

Reply:
Nice catch, somehow I missed it, thanks! waiting for CI to verify